editor.planx.uk/src/@planx/components/ContactInput/Public.tsx

```diff
@@ -0,0 +1,156 @@
+import Card from "@planx/components/shared/Preview/Card";
+import QuestionHeader from "@planx/components/shared/Preview/QuestionHeader";
+import type { PublicProps } from "@planx/components/ui";
+import { useFormik } from "formik";
+import React from "react";
+import Input from "ui/Input";
+import InputLabel from "ui/InputLabel";
+import InputRowItem from "ui/InputRowItem";
+
+import { ERROR_MESSAGE } from "../shared/constants";
+import { getPreviouslySubmittedData, makeData } from "../shared/utils";
+import type { Contact, ContactInput } from "./model";
+import { userDataSchema } from "./model";
+
+export type Props = PublicProps<ContactInput, Contact>;
+
+interface FormProps {
+  title: string;
+  firstName: string;
+  lastName: string;
+  organisation: string;
+  phone: string;
+  email: string;
+  "name.first"?: string;
+  "name.last"?: string;
+  "company.name"?: string;
+  "phone.primary"?: string;
+}
+
+export default function ContactInputComponent(props: Props): FCReturn {
+  const formik = useFormik<FormProps>({
+    initialValues: getPreviouslySubmittedData(props) ?? {
+      title: "",
+      firstName: "",
+      lastName: "",
+      organisation: "",
+      phone: "",
+      email: "",
+    },
+    onSubmit: (values) => {
+      // map internal props to expected granular passport keys before submitting
+      values["name.first"] = values.firstName;
+      values["name.last"] = values.lastName;
+      values["company.name"] = values.organisation;
+      values["phone.primary"] = values.phone;
```

Comment:
this mapping feels particularly clumsy, but `formik.handleChange` breaks on an input if its' direct name includes a `.`

see this attempt: 
![Screenshot from 2022-11-04 08-55-42](https://user-images.githubusercontent.com/5132349/199954923-1ca05c30-f30f-43db-a65c-a79bdf94c43d.png)

then formik failed to reconcile the passport and we had duplicate: `name.first: <>` & `name: { first: <> }` like this which prevented being able to chagne the first name input:
![Screenshot from 2022-11-04 08-54-10](https://user-images.githubusercontent.com/5132349/199955347-dc088eed-2493-4cab-b6d2-215c6cab5796.png)

i fiddled with this for awhile and eventually just settled on this mapping, but maybe i'm forgetting some syntax to accomplish this in a more eloquent way? any better ideas?

Reply:
So it certainly looks like this should be do-able - there's an option in the docs to avoid nesting https://formik.org/docs/guides/arrays#avoid-nesting and a `getIn()` method which allows us to access values/errors with the following syntax - `value={getIn(formik.values, "first.name")}`

Spent a bit of time on it and I can get it all working apart from the Yup Schema which doesn't seem to handle this. I think mapping is the sensible way forward here!

Question - if we're mapping values anyway, is there a way we could retain the original mappings with something like this and remove both complexity elsewhere and (small) issues for reconciling users?
```ts
values[`${parentPassportValue}.name.first`] = values.firstName;
```

Reply:
This is such a finicky topic that I've been feeling stuck & unsatisfied around too - I've hit the same issues with formik & Yup. 

BUT, I think I landed on a better mapping that fits your second question well:

I'm using this representation for the passport - which keeps the existing/expected passport keys in the top level, and adds a `_contact.[props.fn]` nested object to the passport that we can reference internally for back button/change behavior. This feels like a similar patern to FindProperty's `_address` passport field and accounts for the likelihood that we will have multiple ContactInputs per flow.

The tricky bit was writing custom `makeData` & `getPreviouslySubmittedData` methods rather than referencing the shared utils, but I think it's worth it for this passport representation which then negates all those silly `||` conditionals I originally introduced in the Send schemas. Thoughts?

![Screenshot from 2022-11-15 15-18-51](https://user-images.githubusercontent.com/5132349/202001681-59c248ee-0fbe-42c4-8959-c537c7609d73.png)